ecs.tf

```diff
@@ -117,24 +119,97 @@ data "aws_iam_policy_document" "ec2_ecs_task_s3_bucket" {
       "s3:DeleteObject",
     ]
 
-    resources = [
-      "arn:aws:s3:::${aws_s3_bucket.bucket.bucket}",
-      "arn:aws:s3:::${aws_s3_bucket.bucket.bucket}/*"
-    ]
+    resources = concat([
+      for bucketName in var.bucket_names :
+      "arn:aws:s3:::${bucketName}"
+      ], [
+      for bucketName in var.bucket_names :
+      "arn:aws:s3:::${bucketName}/*"
+      ]
+    )
   }
 }
 
+
 resource "aws_iam_role_policy" "ec2_ecs_task_s3_bucket" {
+  count = length(var.bucket_names) == 1 ? 1 : 0
+
   name = "${var.cluster_name}-ecs-task-s3"
   role = aws_iam_role.ecs_task.id
 
-  policy = data.aws_iam_policy_document.ec2_ecs_task_s3_bucket.json
+  policy = data.aws_iam_policy_document.ec2_ecs_task_s3_bucket[0].json
+}
+
+data "aws_iam_policy_document" "ec2_ecs_task_s3_compaction_bucket" {
```

Comment:
can we call this `s3_standard_bucket` instead of `s3_compaction_bucket` for two reasons:

1. The other one is s3express_bucket so this follows that pattern
2. Its not unimaginable that someone would want to use s3express for both ingestion and compaction (it would be very expensive, but could be a low volume use-case where they just want low latency no matter what